docs/next/guides/getting-started/setting-options.md

```diff
@@ -1,239 +1,674 @@
+      height: 300,
+      // the `cells` option
+      cells:
+    });
+    ```
+2. Set the [`cells`](@/api/options.md#cells) option to a function. It can take three arguments:<br>
+   - `row`: a row coordinate (a **physical** index)
+   - `col`: a column coordinate (a **physical** index)
+   - `prop`: if your [`data`](@/api/options.md#data) is an [array of objects](@/guides/getting-started/binding-to-data.md#array-of-objects), `prop` is a property name for a column's data source object.<br>
+   If your [`data`](@/api/options.md#data) is an [array of arrays](@/guides/getting-started/binding-to-data.md#array-of-arrays), `prop` is the same as `col`.
+    ```js
+    const hot = new Handsontable(container, {
+      // the `cells` option set to a function
+      cells: function(row, col, prop) {
+        // the `cells` function's body
+      }
+    });
+    ```
+3. In the [`cells`](@/api/options.md#cells) function's body, implement a logic that selects your required columns, rows, or cells (as combinations of `row` and `col` coordinates).<br>
+   For example:
+    ```js
+    const hot = new Handsontable(container, {
+      cells: function (row, col) {
+        if (row === 1 || row === 5 && col === 1) {
+          return {
+            readOnly: true,
+            className: 'read-only-background'
+          };
+        }
+      }
+    });
+    ```
+
+#### Example
+
+In the example below, the modified [`cells`](@/api/options.md#cells) options overwrite the top-level grid options.
 
 ::: example #example5 --html 1 --css 2 --js 3
 ```html
 <div id="example5" class="hot"></div>
 ```
 ```css
-td.bg-read-only {
+td.read-only-background {
   background-color: #f2f4fb;
 }
 ```
 ```js
 const container = document.querySelector('#example5');
 
 const hot = new Handsontable(container, {
+  // top-level grid options that apply to the entire grid
+  licenseKey: 'non-commercial-and-evaluation',
+  data: Handsontable.helper.createSpreadsheetData(5, 10),
+  width: 'auto',
+  height: 'auto',
+  rowHeaders: true,
+  colHeaders: true,
+  // the `cells` option overwrites the top-level grid options
+  // apply only to cells selected by your custom logic
+  cells: function (row, col) {
+    if (row === 1 || row === 3 && col === 1) {
+      return {
+        readOnly: true,
+        className: 'read-only-background'
+      };
+    }
+  }
+});
+```
+:::
+
+## Configuration example
+
+In the example below, some cells are read-only, and some cells are editable:
+- By default, all cells are read-only (as set in the top-level [grid options](#setting-grid-options)).
+- For the first column, the mid-level [column options](#setting-column-options) overwrite the top-level [grid options](#setting-grid-options).<br>
+  As a result, the first column cells are editable.
+- For cell `A1` (`0`, `0`), the bottom-level [cell options](#setting-cell-options) overwrite both the mid-level [column options](#setting-column-options), and the top-level [grid options.](#setting-grid-options)<br>
```

Comment:
Here is the same situation as the one above we could change the 

> For cell `A1` (`0`, `0`),

to

> For cell `A1` `(0, 0)`,

it is definitely easier to read